src/Swashbuckle.AspNetCore.Filters/Examples/MvcOutputFormatter.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Formatters;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Net.Http.Headers;
+using System.Linq;
+
+namespace Swashbuckle.AspNetCore.Filters
+{
+    internal class MvcOutputFormatter
+    {
+        private readonly object initializeLock = new object();
+        private bool initializedOutputFormatterSelector;
+
+        private readonly IOptions<MvcOptions> selectorOptions;
+        private readonly ILoggerFactory loggerFactory;
+
+        private OutputFormatterSelector outputFormatterSelector;
+        private OutputFormatterSelector OutputFormatterSelector
+        {
+            get
+            {
+                lock (initializeLock)
+                {
+                    if (!initializedOutputFormatterSelector)
+                    {
+                        if (selectorOptions != null)
+                        {
+                            outputFormatterSelector = new DefaultOutputFormatterSelector(
+                                selectorOptions,
+                                loggerFactory ?? throw new ArgumentNullException(nameof(loggerFactory)));
+                        }
+                        initializedOutputFormatterSelector = true;
+                    }
+                }
+
+                return outputFormatterSelector;
+            }
+        }
+
+        public MvcOutputFormatter(IOptions<MvcOptions> options, ILoggerFactory loggerFactory)
+        {
+            this.initializedOutputFormatterSelector = false;
+            this.selectorOptions = GetSelectorOptions(options);

```

Comment:
Can this be called only when needed, rather than in the constructor?

My concern is that 99% of users of this library use JSON only and not XML, so I don't want their applications to go through any of the XML logic unless they are going to use it.

Reply:
Yes, @ridingwolf can we move this also in a lazy loaded way please?